tests/lib/cascading-config-array-factory.js

```diff
@@ -137,10 +176,13 @@ describe("CascadingConfigArrayFactory", () => {
             });
 
             describe("when '~/.eslintrc.json' exists and CWD is `~/`", () => {
-                beforeEach(() => {
-                    cwd = homeDir;
-                    const { CascadingConfigArrayFactory } = defineCascadingConfigArrayFactoryWithInMemoryFileSystem({
-                        cwd: () => cwd,
+
+                let prepare, cleanup, getPath;
+                
+                beforeEach(async () => {
+
+                    ({ prepare, cleanup, getPath } = createCustomTeardown({
+                        cwd: homeDir,
```

Comment:
Since this uses the actual home directory now, some of these tests could fail if someone already has a `~/.eslintrc.json` file there. Thankfully the teardown library appears to leave the file alone if it already exists, so at least it's not destructive:

```sh
$ echo '!@#$%^&*()' > ~/.eslintrc.json
$ npm test
# failures
$ cat ~/.eslintrc.json
!@#$%^&*()
```

Unless you have a bright idea for a way to test this _without_ touching the home directory, I wouldn't be totally opposed to skipping the home directory tests outside CI. It's a little weird, but ideally people won't be touching the code in this repository anymore once you're done with the migration.

Reply:
I'd say at this point let's just let the weird tests be weird. I can't wrap my brain around most of what the tests are doing at this point, so I'd call it a success when stuff starts passing even if it's not ideal.

Reply:
Looks like this uses a fake home dir, created in the project's parent dir:

https://github.com/eslint/eslintrc/blob/0acc01809e790d2c7f2434d90fe011d7bf0f5048/tests/lib/cascading-config-array-factory.js#L166-L172

Should we maybe avoid creating and deleting (potentially already existing and unrelated) directories there, by setting `homeDir` to a path inside `systemTempDir`?

Reply:
I really don't want to be changing how the existing tests work. I've had a hard enough time converting these tests, so I'd rather just keep everything as close to original as possible.